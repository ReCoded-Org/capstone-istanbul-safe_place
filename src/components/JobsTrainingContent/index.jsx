import React, { useState } from "react";
import { Container, Row, Col } from "react-bootstrap";
import "./index.scss";
import { useTranslation } from "react-i18next";
import OpportunitiesCard from "../OpportunitiesCard";
import SearchBar from "../SearchBar";
import JobsTrainingButtons from "../JobsTrainingButtons";

const JobsTrainingContent = () => {
  const { t } = useTranslation();
  const [cardsArr, setCardsArr] = useState([
    ...t(`home.jobOpportunities`, { returnObjects: true }),
  ]);

  const [searchCard, setSearchCards] = useState(cardsArr);

  const handleSearch = (e, keyword) => {
    e.preventDefault();
    setSearchCards(cardsArr);
    setSearchCards((prevState) => {
      const searchedCards = [...prevState];
      return searchedCards.filter((card) =>
        card.title.toLowerCase().includes(keyword)
      );
    });
  };

  return (
    <Container className="jobsTrainingContent">
      <Row className="searchBar">
        <SearchBar handleSearch={handleSearch} />
      </Row>
      <Row>
        <Col md={3}>
          <JobsTrainingButtons
            setSearchCards={setSearchCards}
            setCardsArr={setCardsArr}
          />
        </Col>
        <Col>
          <Row className="cards">
            {searchCard.map((opportunity, index) => (
              <Col
                sm={12}
                md={6}
                className={index % 2 === 0 ? "peachBorder" : "greenBorder"}
              >
                <OpportunitiesCard {...opportunity} key={opportunity.id} />
              </Col>
            ))}
          </Row>
        </Col>
      </Row>
    </Container>
  );
};

export default JobsTrainingContent;
