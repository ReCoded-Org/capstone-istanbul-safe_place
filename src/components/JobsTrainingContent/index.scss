@import "../../style/variables";
@import "../../style/buttons";

.jobsTrainingContent {
  margin-top: 3rem;

  .trainingsBtn,
  .jobsBtn {
    @include primaryBtn;
    width: 12.75rem;
    border-radius: 0;
    text-align: right;
    color: $darkColor;
    background-color: transparent;

    &:focus,
    &:active,
    &:hover,
    &:disabled {
      border-right: solid $primaryColor;
      outline: 0;
      color: $primaryColor;
      background-color: transparent;
    }
  }

  .cards {
    width: 100%;
  }

  .greenBorder > * {
    border: solid 0.15rem $secondaryColor;

    .title {
      color: $secondaryColor;
    }

    .borderLine {
      border-bottom: 0.125rem solid $secondaryColor;
    }
  }

  .peachBorder > * {
    border: solid 0.15rem $warningColor;
  }
}
