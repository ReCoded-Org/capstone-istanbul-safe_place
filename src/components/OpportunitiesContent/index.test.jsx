import React from "react";
import renderer from "react-test-renderer";
import JobsTrainingContent from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<JobsTrainingContent />).toJSON();
  expect(tree).toMatchSnapshot();
});
