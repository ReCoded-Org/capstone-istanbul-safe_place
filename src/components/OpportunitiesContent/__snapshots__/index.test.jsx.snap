// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="jobsTrainingContent container"
>
  <div
    className="searchBar row"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <input
            className="mr-sm-2 searchBarInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search"
          />
          <img
            alt="Search Icon"
            className="searchIcon"
            onClick={[Function]}
            src="searchIcon.svg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <div
        className="opportunityBtns"
      >
        <button
          className="opportunityListBtn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          jobsTrainingPage.buttons.jobOpportunities
        </button>
        <button
          className="opportunityListBtn"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          jobsTrainingPage.buttons.trainings
        </button>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="cards row"
      >
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="greenBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
        <div
          className="peachBorder col-md-6 col-sm-12"
        >
          <div
            className="opportunities col"
          >
            <div
              className="postDate"
            />
            <div
              className="title"
            />
            <div
              className="description"
            />
            <span
              className="borderLine"
            />
            <div
              className="contactsDetails"
            >
              <div
                className="address"
              />
              <div
                className="nameOfOpportunityOwner"
              />
              <div
                className="phoneNumber"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
